### ISSUE TITLE
Flaky test - `TestSourcesFromTestdataWithIssuesDir/staticcheck.go`

### ISSUE THREAD

**Issue by butuzov** [golangci/golangci-lint#2296]
### Description of the problem

CI tests are constantly failing if run on UNIX on this test (it can pass on 1.15/1.16/1.17, but fail on any of these versions)

````
2021-10-18T13:01:14.0231105Z === CONT  TestSourcesFromTestdataWithIssuesDir/staticcheck.go
2021-10-18T13:01:14.0232163Z     linters_test.go:36: 
2021-10-18T13:01:14.0232733Z         	Error Trace:	linters_test.go:36
2021-10-18T13:01:14.0233386Z         	            				linters_test.go:157
2021-10-18T13:01:14.0234053Z         	            				linters_test.go:56
2021-10-18T13:01:14.0234708Z         	Error:      	Received unexpected error:
2021-10-18T13:01:14.0235752Z         	            	staticcheck.go:26: missing error "SA1019: runtime.CPUProfile has been deprecated .*"
2021-10-18T13:01:14.0237143Z         	Test:       	TestSourcesFromTestdataWithIssuesDir/staticcheck.go
```

to research
